src/lib/pdfconverter.cc

```diff
@@ -890,10 +925,33 @@ void PdfConverterPrivate::beginPrintObject(PageObject & obj) {
 	if (!obj.loaderObject || obj.loaderObject->skip)
 		return;
 
-	QWebPrinter *webPrinter = objects[currentObject].web_printer;
-	if (webPrinter == 0)
-		webPrinter = objects[currentObject].web_printer = \
-			new QWebPrinter(obj.page->mainFrame(), printer, *painter);
+//	QWebPrinter *webPrinter = objects[currentObject].web_printer;
+//	if (webPrinter == 0)
+//		webPrinter = objects[currentObject].web_printer = \
+//			new QWebPrinter(obj.page->mainFrame(), printer, *painter);
+
+// Need to set the Orientation of the Page again before generating the Output
+	if (obj.settings.pageSpecificOrientation != NULL){
+
+			QChar c = obj.settings.pageSpecificOrientation.at(0);
+			char orientParam = c.toUpper().toLatin1();
+      		//fprintf(stdout, "Custom page orientation (in beginPrntObject) Page %d ",currentObject);	
+
+			if (orientParam == 'L'){
+				printer->setOrientation(QPrinter::Landscape);
+		      //fprintf(stdout, "Landscape\n");	
+				}
+				else{
+				printer->setOrientation(QPrinter::Portrait);
+		      //fprintf(stdout, "Portrait\n");	
+				}			
+			}
+
+// With the above original code the web printer if !=0 would not hold the Portrait or Landscape resetting
+// Replaced with the below - a forced new webPrinter
+  QWebPrinter *webPrinter = objects[currentObject].web_printer;
+	webPrinter = objects[currentObject].web_printer = new QWebPrinter(obj.page->mainFrame(), printer, *painter);
```

Comment:
Consider replacing lines 952 and 953 with the below code so that we only force a new webPrinter if there is none or we have a request to change the orientation. This would align better to the original code.

```
  QWebPrinter *webPrinter = objects[currentObject].web_printer;

	if (webPrinter == 0 || obj.settings.pageSpecificOrientation != NULL)
	webPrinter = objects[currentObject].web_printer = \
		new QWebPrinter(obj.page->mainFrame(), printer, *painter);
```